components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -27,16 +27,14 @@ internal const val CACHE_FILE_NAME = "mozilla_components_service_contile.json"
 internal const val MINUTE_IN_MS = 60 * 1000
 
 /**
- * Provide access to the Contile services API.
+ * Provides access to the Contile services API.
  *
- * @property context A reference to the application context.
  * @property client [Client] used for interacting with the Contile HTTP API.
  * @property endPointURL The url of the endpoint to fetch from. Defaults to [CONTILE_ENDPOINT_URL].
  * @property maxCacheAgeInMinutes Maximum time (in minutes) the cache should remain valid
  * before a refresh is attempted. Defaults to -1, meaning no cache is being used by default.
  */
 class ContileTopSitesProvider(
-    private val context: Context,
```

Comment:
We need to make `ContilieTopSitesProvider` accessible statically for worker, but having a reference to a class with `Context` causes a memory leak according to the linter.

Reply:
If I understood correctly the issue is because the lint tool is complaining because we are implicitly keeping a context in  static reference, via the [ContileTopSitesUseCases.provider](https://github.com/mozilla-mobile/android-components/pull/11721/files#diff-ef36047fec122291ca0dc4cac399d07cb889945f546448e2434a8622df932198R33) class , right?

If that it's case, I think as long as we are using an `applicationContext` we should be safe from a memory leak,  maybe we could explicitly call `context = context.getApplicationContext()` which could make the linter happy, and rename the `context` property to `applicationContext` to avoid any misunderstanding.  What do you think? :)

Reply:
Can give this as try